src/clib/pioc_sc.c

```diff
@@ -261,28 +277,34 @@ int CalcStartandCount(int basetype, int ndims, const int *gdims, int num_io_proc
         piodie("Invalid basetype ",__FILE__,__LINE__);
         break;
     }
+
+    /* Determine the minimum block size. */
     minblocksize = minbytes / basesize;
 
+    /* Find the total size of the data. */
     pgdims = 1;
     for (i = 0; i < ndims; i++)
         pgdims *= (long int)gdims[i];
+
+    /* ??? */
     p = pgdims;
     use_io_procs = max(1, min((int)((float)p / (float)minblocksize + 0.5), num_io_procs));
     converged = 0;
     for (i = 0; i < ndims; i++)
     {
         mystart[i] = 0;
-        mykount[i] = 0;
+        mycount[i] = 0;
     }
 
+    /* ??? */
```

Comment:
use_io_procs is the number of ioprocs that are needed so that we have blocksize data on each iotask, now find start and count values needed on each of these tasks.